zxcvbn-test/ZxcvbnTestMesserliExtended.cs

```diff
@@ -0,0 +1,138 @@
+            var ABab = passwordChecker.EvaluatePassword("ABab");
+
+            Assert.IsTrue(aaaa.Entropy > a.Entropy);
+            Assert.IsTrue(abab.Entropy > aaaa.Entropy);
+            Assert.IsTrue(ABab.Entropy > abab.Entropy);
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void DictionaryMatcherTest()
+        {
+            // We test if the dictionary matcher loads the messerli dictionary
+            var passwordChecker = new Zxcvbn.Zxcvbn();
+            var password = "Messerli";
+            var metric = passwordChecker.EvaluatePassword(password);
+
+            foreach (var match in metric.MatchSequence)
+            {
+                // Match
+                Assert.AreEqual(10, match.Cardinality);
+                Assert.AreEqual(password, match.Token);
+                Assert.AreEqual(0, match.i);
+                Assert.AreEqual(password.Length - 1, match.j);
+
+                // Dictionary Match
+                var dictionaryMatch = match as Zxcvbn.Matcher.DictionaryMatch;
+
+                Assert.AreNotEqual(null, dictionaryMatch);
+                Assert.AreEqual("messerli", dictionaryMatch.DictionaryName);
+                Assert.AreEqual(1, dictionaryMatch.UppercaseEntropy);
+
+
+            }
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void SwissSpatialMatcherTest()
+        {
+            // We test if the pattern matcher matches swiss keyboards
+            var swissKeyboardPatterns = new System.Collections.Generic.List<string> { "löäü'", "ö-.,m" };
+
+            var passwordChecker = new Zxcvbn.Zxcvbn();
+
+            foreach (var pattern in swissKeyboardPatterns)
+            {
+                var metric = passwordChecker.EvaluatePassword(pattern);
+
+                foreach (var match in metric.MatchSequence)
+                {
+                    // Dictionary Match
+                    var spatialMatch = match as Zxcvbn.Matcher.SpatialMatch;
+
+                    Assert.AreNotEqual(null, spatialMatch);
+                    Assert.AreEqual("Swiss German", spatialMatch.Graph);
+                    Assert.AreEqual(0, spatialMatch.ShiftedCount);
+                    Assert.AreEqual(2, spatialMatch.Turns);
+
+                    // Match
+                    Assert.AreEqual(0, match.Cardinality);
+                    Assert.AreEqual(0, match.i);
+                    Assert.AreEqual(pattern.Length - 1, match.j);
+                    Assert.AreEqual(pattern, match.Token);
+                }
+            }
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void FrenchSpatialMatcherTest()
+        {
+            // We test if the pattern matcher matches swiss keyboards
+            var frenchKeyboardPatterns = new System.Collections.Generic.List<string> { "*ùmlkj", "bn,;:!" };
```

Comment:
Import this type.

Reply:
Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/d1c2b15d6614e86b295ed96549f592061148e085

Reply:
Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/49b4d6adbe9b58c32c782b87f43946d10f108e6f